LuckyDragon/app/src/main/java/com/example/luckydragon/Activities/ViewEventActivity.java

```diff
@@ -57,8 +56,11 @@ protected void onCreate(Bundle savedInstanceState) {
         event.fetchData(); // get all event data
         viewEventView = new ViewEventView(event, this);
 
-        // Hide buttons for entrant
-        if(globalApp.getRole() == GlobalApp.ROLE.ENTRANT) {
+        // Hide buttons for entrant or if event does not have QR Code
+        if (globalApp.getRole() == GlobalApp.ROLE.ENTRANT) {
+            hideQrCodeButton();
+        }
+        if (event.getQRBitMatrix() == null) {
```

Comment:
hiding and showing logic should be in the view (might need to pass the role to the view)

I think this causes the bug where if even if the qr is empty, there still sometimes might be the qr button. This happens on reboot in tandem with the bug about removing the qr on reboot.

Reply:
yeah I moved the logic inside the view